src/models/hit-types/text-hit.ts

```diff
@@ -0,0 +1,198 @@
+    return this.rawMetadata?.fields?.downloads
+      ? new NumberField(this.rawMetadata.fields.downloads)
+      : undefined;
+  }
+
+  @Memoize() get filename(): StringField | undefined {
+    return this.rawMetadata?.fields?.filename
+      ? new StringField(this.rawMetadata.fields.filename)
+      : undefined;
+  }
+
+  @Memoize() get file_basename(): StringField | undefined {
+    return this.rawMetadata?.fields?.file_basename
+      ? new StringField(this.rawMetadata.fields.file_basename)
+      : undefined;
+  }
+
+  @Memoize() get file_creation_mtime(): NumberField | undefined {
+    return this.rawMetadata?.fields?.file_creation_mtime
+      ? new NumberField(this.rawMetadata.fields.file_creation_mtime)
+      : undefined;
+  }
+
+  @Memoize() get mediatype(): MediaTypeField | undefined {
+    return this.rawMetadata?.fields?.mediatype
+      ? new MediaTypeField(this.rawMetadata.fields.mediatype)
+      : undefined;
+  }
+
+  /** Optional. */
+  @Memoize() get page_num(): NumberField | undefined {
+    return this.rawMetadata?.fields?.page_num
+      ? new NumberField(this.rawMetadata.fields.page_num)
+      : undefined;
+  }
+
+  /** Optional. */
+  @Memoize() get publicdate(): DateField | undefined {
+    return this.rawMetadata?.fields?.publicdate
+      ? new DateField(this.rawMetadata.fields.publicdate)
+      : undefined;
+  }
+
+  /**
+   * Computed in processing of FTS API hit.
+   */
+  @Memoize() get result_in_subfile(): BooleanField | undefined {
+    return this.rawMetadata?.fields?.result_in_subfile != null
+      ? new BooleanField(this.rawMetadata.fields.result_in_subfile)
+      : undefined;
+  }
+
+  /** Optional. */
+  @Memoize() get reviewdate(): DateField | undefined {
+    return this.rawMetadata?.fields?.reviewdate
+      ? new DateField(this.rawMetadata.fields.reviewdate)
+      : undefined;
+  }
+
+  /**
+   * Optional.
+   * Multivalued.
+   */
+  @Memoize() get subject(): StringField | undefined {
+    return this.rawMetadata?.fields?.subject
+      ? new StringField(this.rawMetadata.fields.subject)
+      : undefined;
+  }
+
+  /** Optional. */
+  @Memoize() get title(): StringField | undefined {
+    return this.rawMetadata?.fields?.title
+      ? new StringField(this.rawMetadata.fields.title)
+      : undefined;
+  }
+
+  @Memoize() get updated_on(): DateField | undefined {
+    return this.rawMetadata?.fields?.updated_on
+      ? new DateField(this.rawMetadata.fields.updated_on)
+      : undefined;
+  }
+
+  /**
+   * Computed from date.
+   * Optional.
+   */
+  @Memoize() get year(): NumberField | undefined {
+    return this.rawMetadata?.fields?.year
+      ? new NumberField(this.rawMetadata.fields.year)
+      : undefined;
+  }
+
+  /**
+   * Synthesized in processing of FTS API hit; TBD
+   * Optional.
+   */
+  @Memoize() get __href__(): StringField | undefined {
+    return this.rawMetadata?.fields?.__href__
+      ? new StringField(this.rawMetadata.fields.__href__)
+      : undefined;
+  }
+}
```

Comment:
QUESTION: It sounds like the two return objects from the search engines were different enough to warrant two completely different types. I'd be interested in knowing more about this via a real-time conversation. I'd have hoped that the PageProductionService would hide these differences and just return a single kind of search results.

Reply:
Agreed, let's discuss this at tomorrow's meeting. Quick summary is that the PPS responses themselves do have a consistent overall structure, with the exception of the search hits themselves. The hits for Metadata Search (`item` hits) and Full Text Search (`text` hits) currently conform to two different schemas. The corresponding schema is currently provided alongside each response, though that might change as we evolve out of beta.

The search service returns hits typed uniformly as a high-level `Hit` interface that tries to hide this distinction from callers in a type-safe fashion. `Hit` includes the properties from both hit types, with the expectation that callers will be handling possible `undefined` property values anyway due to so many properties being optional. But maybe there's a more elegant way to address this...